File: lib/BSI/BSIFilter.php
<?php
require_once("JSONRPC/ClientFilterBase.php");
require_once("BSI/BSI_Exception.php");


class BSIFilter extends JSONRPC_client_filter_plugin_base
{

	public function __construct()
	{

	}

    public function exceptionCatch($exception)
	{
        if($exception->getCode() >= 0)
            throw new BSI_Exception($exception->getMessage(), $exception->getCode());
        else 
            throw $exception;
    }
}

File: lib/BSI/BSIFilter.py
from JSONRPC.ClientFilterBase import JSONRPC_client_filter_plugin_base
from BSI_Exception import BSI_Exception

class BSIFilter(JSONRPC_client_filter_plugin_base):

	def exceptionCatch(self, exception):

		if exception.nCode >= 0:
			raise BSI_Exception(exception.strMessage, exception.nCode)
		else:
			raise exception
